pinot-query-planner/src/main/java/org/apache/calcite/jdbc/CalciteSchemaBuilder.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.calcite.jdbc;
+
+import org.apache.calcite.schema.Schema;
+
+
+/**
+ * This class is used to create a {@link CalciteSchema} with a given {@link Schema} as the root.
+ *
+ * <p>This class resides in calcite.jdbc namespace because there's no complex logic we have in terms of catalog-based
+ * schema construct. We instead create a {@link SimpleCalciteSchema} that's package protected.
+ */
+public class CalciteSchemaBuilder {
```

Comment:
Can you add some notes / javadoc on why we need to create a wrapper CalciteSchema on top of another Calcite schema object ? 

From the [design doc](https://docs.google.com/document/d/1Vbc4MRFzkeGKaYxvVctYUlj9f1P7DfzmoyrNnFIUnmE/edit#), I understand why we need this to interface with Calcite concept of catalog/schema but shouldn't Pinot native Schema be plugged into this ?

Reply:
yes but we have to conformed with calcite's plugin which requires a CalciteSchema class. but since our schema objects are so simple we dont have to create our own CalciteSchema and overwrite so I decided to use the SimpleCalciteSchema (which is protected access)

I am open to copy over the impl and renaming it to PinotCalciteSchema but that will cause unsync changes issue